### PR TITLE
MF-1329 - Add Python SDK

### LINKED ISSUE #1329: Create Python SDK
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
We currently provide Go SDK. It would be useful to provide Python SDK as well, to facilitate creating Python apps on the top of Mainflux.
 
3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
nice-to-have

### PR DESCRIPTION
**Pull request by fbugarski**
Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
This pull request adds Python SDK.

### Which issue(s) does this PR fix/relate to?
Resolves #1329 

### List any changes that modify/break current functionality
Does not affect any current functionality.  

### Have you included tests for your changes?
Tests are not currently included, they will be added subsequently.

### Did you document any new/modified functionality?
Documentation will be added.

### CODE REVIEW

File: pkg/sdk/python/channels.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+def CreateChannel(channel, token):
+    resp = requests.post("http://localhost/channels", json=channel, headers={"Authorization": token})
+    return resp.headers.get("location")
+
+def CreateChannels(channels, token):
+    resp = requests.post("http://localhost/channels", json=channels, headers={"Authorization": token})
+    return resp
+
+def Channels(params, token):
+    url = "http://localhost/channels"+'?'+'offset='+params['offset']+'&'+'limit='+params['limit']+'&'+'connected='+params['connected']
```
Comment (manuio) [inline pkg/sdk/python/channels.py]: What if `params['offset']` is empty?

File: pkg/sdk/python/channels.py [hunk 2]
```diff
@@ -20,13 +20,13 @@ def channels(self, query_params, token):
         return resp.json()
 
     def channel_by_thing(self, thing_id, query_params, token):
```
Comment (manuio) [inline pkg/sdk/python/channels.py]: This returns a list of channels so better call it `channels_by_thing` or `list_chanels_by_thing`

File: pkg/sdk/python/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
```
Comment (drasko) [inline pkg/sdk/python/LICENSE.md]: No need for this license, we already have a license in the top-level Mainflux repo

File: pkg/sdk/python/tests/test_version.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import sdk
+import pytest
+import json
+
+s = sdk.SDK()
+
```
Comment (drasko) [inline pkg/sdk/python/tests/test_version.py:6]: Kill these whitelines

File: pkg/sdk/python/setup.cfg [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+project_urls =
+    Bug Tracker = https://github.com/mainflux/mainflux/tree/master/pkg/sdk/python
+classifiers =
+    Programming Language :: Python :: 3.8.5
+    License :: OSI Approved :: MIT License
+    Operating System :: OS Independent
+
+[options]
+package_dir =
+    = src
```
Comment (drasko) [inline pkg/sdk/python/setup.cfg]: This is strange - why not in the same line?

File: pkg/sdk/python/functions/examples.py [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+thing_id = "02dc3c7c-e33b-471d-9ab1-ba49c7c8015b"
+user_id = "2803446a-4109-42e5-b07c-696e680294c8"
+user = {"email": "FB123@example.com", "password": "12345678"}
+group_id = "01F1JZ23RP7VW60MKAWEF7QKG9"
+token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTY2MjIwODMsImlhdCI6MTYxNjU4NjA4MywiaXNzIjoibWFpbmZsdXguYXV0aCIsInN1YiI6InBvc2xlZG5qaUBleGFtcGxlLmNvbSIsImlzc3Vlcl9pZCI6IjI4MDM0NDZhLTQxMDktNDJlNS1iMDdjLTY5NmU2ODAyOTRjOCIsInR5cGUiOjB9.xDWdnqwivjJnrSt_pNnwbIpWRfVwg8apxf_GNidgmFw"
+default_url = "http://localhost"
+
+sdk = SDK()
+
+# '''To start working with the Mainflux system, you need to create a user account'''
```
Comment (drasko) [inline pkg/sdk/python/functions/examples.py]: This will be needed to be uncommented in the end

**Review by drasko — CHANGES_REQUESTED** [review]

File: pkg/sdk/python/setup.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import setuptools
+
+with open("README.md", "r", encoding="utf-8") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="example-pkg-fbugarski",
```
Comment (drasko) [inline pkg/sdk/python/setup.py]: Name of the SDK is probably `mainflux`

**Review by drasko — CHANGES_REQUESTED** [review]

File: pkg/sdk/python/setup.py [hunk 8]
```diff
@@ -17,7 +17,6 @@
     },
     classifiers=[
         "Programming Language :: Python :: 3.8.5",
-        "License :: OSI Approved :: MIT License",
```
Comment (manuio) [inline pkg/sdk/python/setup.py]: Maybe better to define the licence?

File: pkg/sdk/python/functions/examples.py [hunk 9]
```diff
@@ -1,24 +1,25 @@
...
-import channels
-import groups
+from functions import users
+from functions import things
+from functions import messages
+from functions import channels
+from functions import groups
 import requests
-from sdk import SDK
+from functions import SDK
```
Comment (manuio) [inline pkg/sdk/python/functions/examples.py:7]: This sounds too generic as naming

**Comment by drasko** [comment]
Closing in favor of https://github.com/mainflux/sdk-py/pull/1 - Python SDK will live in a separate dedicated repo.

**Pull request closed without merging**